x-pack/platform/plugins/shared/onechat/server/services/tools/persisted/tool_types/index_search/validate_configuration.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClient } from '@kbn/core/server';
+import type { IndexSearchToolConfig } from '@kbn/onechat-common/tools';
+import { createBadRequestError } from '@kbn/onechat-common';
+import { listSearchSources } from '@kbn/onechat-genai-utils';
+
+const CCS_TOKEN = ':';
+
+export const validateConfig = async ({
+  config,
+  esClient,
+}: {
+  config: IndexSearchToolConfig;
+  esClient: ElasticsearchClient;
+}) => {
+  const { pattern } = config;
+  if (pattern.includes(CCS_TOKEN)) {
+    throw createBadRequestError(`Cross-cluster search is not supported by the index_search tool`);
+  }
```

Comment:
We block CCS patterns for now, to avoid opening that can of worms. 

We can revisit later if we want to, but it's not worth managing all the edge cases and limitations today (can't get mappings via CCS for example today)